src/app/components/hris/sign-in/sign-in.component.ts

```diff
@@ -107,7 +103,24 @@ export class SignInComponent {
                 sameSite: 'None'
               });
   
-              return { user, token, role: filteredRoles };
+              if (filteredRoles.length === 0) {
+                return this.authService.checkUserExistenceInDatabase().pipe(
+                  switchMap(response => {
+                    if (response === 'User found.') {
+                      this.snackBarService.showSnackbar("Account finalized. You may now log in.", "snack-success");
+                    } else {
+                      this.snackBarService.showSnackbar("Contact admin regarding your account.", "snack-error");
+                    }
+                    return EMPTY;
+                  }),
+                  catchError(() => {
+                    this.snackBarService.showSnackbar("Contact admin regarding your account.", "snack-error");
+                    return EMPTY;
+                  })
+                );
+              }
+  
```

Comment:
```suggestion
```

Reply:
@tbddorothy This change will be a compile error.